### ISSUE TITLE
Sequential Running of Tasks from Task Runner

### ISSUE THREAD

**Issue by GeorgeWL** [Microsoft/vscode#53339]
It would be really nice if you could chain tasks, so you select two tasks, in order, the first finished, the second auto-starts.

I'm aware this can be done through just creating a new task on the `package.json` (and other language equivalents) using a module which does this for you, but it would be nice to have this for those times where you don't want to do so.

**Comment by dbaeumer** [comment]
This is possible using the "dependsOn" property on a task. Unfortunately it is missing documentation but an example is here: 

```
	"tasks": [
		{
			"label": "compile",
			"dependsOn": [
				"compile:server",
				"compile:client"
			],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"problemMatcher": []
		},
		{
			"label": "compile:types",
			"type": "shell",
			"command": "npm run compile:types",
			"group": "build",
			"presentation": {
				"panel": "dedicated",
				"reveal": "never"
			},
			"problemMatcher": [
				"$tsc"
			]
		},
		{
			"label": "compile:jsonrpc",
			"type": "shell",
			"command": "npm run compile:jsonrpc",
			"group": "build",
			"presentation": {
				"panel": "dedicated",
				"reveal": "never"
			},
			"problemMatcher": [
				"$tsc"
			]
		},
		{
			"label": "compile:protocol",
			"type": "shell",
			"command": "npm run compile:protocol",
			"group": "build",
			"presentation": {
				"panel": "dedicated",
				"reveal": "never"
			},
			"problemMatcher": [
				"$tsc"
			],
			"dependsOn": [
				"compile:types",
				"compile:jsonrpc"
			]
		},
		{
			"label": "compile:server",
			"type": "shell",
			"command": "npm run compile:server",
			"group": "build",
			"presentation": {
				"panel": "dedicated",
				"reveal": "never"
			},
			"problemMatcher": [
				"$tsc"
			],
			"dependsOn": [
				"compile:types",
				"compile:jsonrpc",
				"compile:protocol"
			]
		},
		{
			"label": "compile:client",
			"type": "shell",
			"command": "npm run compile:client",
			"group": "build",
			"presentation": {
				"panel": "dedicated",
				"reveal": "never"
			},
			"problemMatcher": [
				"$tsc"
			],
			"dependsOn": [
				"compile:types",
				"compile:jsonrpc",
				"compile:protocol"
			]
		}
    ]

```

**Comment by dbaeumer** [comment]
I do have an issue to add doc for this feature.

**Comment by busticated** [comment]
@dbaeumer - are you saying that the order of `dependsOn` is respected and tasks are run sequentially just based on that field?  [the docs](https://code.visualstudio.com/docs/editor/tasks#_compound-tasks) seem to contradict this:

> ...you can create a task that starts both build scripts in separate terminals

i also have a compound task which currently runs in parallel (vscode v1.25.1)

**Comment by GeorgeWL** [comment]
I'd say it's still worth keeping open then, cause we need to:

1. Rethink the method and maybe further subdivide it.
1. Make it more explicit what it does by adding Docs.
